types/src/shared/utils/hashing.ts

```diff
@@ -3,3 +3,22 @@ import crypto from "crypto";
 export function md5(str: string): string {
   return crypto.createHash("md5").update(str).digest("hex");
 }
+
+function saltedKey(key: string, size = 32): string {
+  const { DUST_DEVELOPERS_SECRETS_SECRET } = process.env;
+  return crypto.createHash('sha256').update(DUST_DEVELOPERS_SECRETS_SECRET + key).digest('base64').substring(0, size);
+}
+
+export function encrypt(text: string, key: string): string {
```

Comment:
Let's add the workspace here

Reply:
Not sure what you mean, the workspace is what's passed as (key) when using the `encrypt()` function [here](https://github.com/dust-tt/dust/pull/4925/files#diff-a58db164ebc85609973763d81dc7f04e733284267d9a2315919b7eb97dee4394R102)?

Reply:
ack didn't get that :+1: Just want to make sure we're salting by both the secret and the workspace :+1: